### PR TITLE
[scripts/release] Add explanations for each section of the release notes.

### PR DESCRIPTION
**Pull request by jverkoey**
The release engineer is expected to fill these sections in by gathering the information from the commits. If information is not available in a given commit, then the release engineer is expected to reach out to the commit author and ask them to provide the additional information (and encouraging them to do so in their commits in the future).

### CODE REVIEW

File: scripts/release [hunk 1]
```diff
@@ -154,9 +154,13 @@ cut_release() {
     echo -e "\n## Component changes" | cat - "$rootdir/CHANGELOG.md" > /tmp/out && mv /tmp/out "$rootdir/CHANGELOG.md"
     cat "$CHANGELOG_PATH" | cat - "$rootdir/CHANGELOG.md" > /tmp/out && mv /tmp/out "$rootdir/CHANGELOG.md"
     echo -e "## API changes" | cat - "$rootdir/CHANGELOG.md" > /tmp/out && mv /tmp/out "$rootdir/CHANGELOG.md"
+    echo -e "Replace this text with example code for each new feature." | cat - "$rootdir/CHANGELOG.md" > /tmp/out && mv /tmp/out "$rootdir/CHANGELOG.md"
```
Comment (romoore) [inline scripts/release:157]: Given the "don't block the release" guidance in the releasing docs, would not having these sections before the release is ready be a sufficient condition for rolling them back?  If the author of a change is OOO and there isn't example code ready to go, should we just roll-back that commit?

Comment (jverkoey) [reply scripts/release:157]: That's a super good question. I'm inclined to say yes so that we encourage better commit descriptions.

Comment (romoore) [reply scripts/release:157]: Great! As a release cop, I don't think I could write good example code for changes I don't know much about.

**Review by romoore — APPROVED** [review]

File: scripts/release (hunk 1 above)
Comment (jverkoey) [reply scripts/release:157]: Agreed.

**Comment by ianegordon** [comment]
This is moving us further away from a completely automated release process. :(

**Review by ianegordon — DISMISSED** [review]
I'm opposed to any changes that put additional burden on the release engineer or require additional human intervention.
I strongly advocate for a process that can be completely scriptable / automated.

**Comment by jverkoey** [comment]
The work being added here is on the authors of commits and pull requests. If/when we add new features, introduce breaking changes, or otherwise land code that might have an impact on clients, the expectation is that the author of that change will communicate the necessary information for the release engineer. As @romoore mentioned, not providing this information is grounds for pulling the change out of the release.

**Comment by ianegordon** [comment]
Perhaps I mis-understand how this is intended to work, but I expect this change will cause a lot of the following in the auto-generated CHANGELOG.md:
"Replace this text with X, Y, or Z"

The release cop may not have insight into all of the details of the PRs.  Additionally, this replacement requires human intervention which moves us further away from automation.

**Review by ianegordon — DISMISSED** [review]
I'm opposed to this change as it adds additional process  and human intervention to an already complex release process.

**Comment by jverkoey** [comment]
The only work the release engineer takes on here is gathering the communication that should already have been written by the author of the relevant changes.

**Comment by jverkoey** [comment]
Gentle ping to @ianegordon.

**Pull request merged**